agent/engine/docker_task_engine.go

```diff
@@ -1230,6 +1212,25 @@ func (engine *DockerTaskEngine) GetTaskByArn(arn string) (*apitask.Task, bool) {
 	return engine.state.TaskByArn(arn)
 }
 
+func (engine *DockerTaskEngine) GetDaemonTask(daemonName string) *apitask.Task {
+	engine.daemonTasksLock.RLock()
+	defer engine.daemonTasksLock.RUnlock()
+	if daemon, ok := engine.daemonTasks[daemonName]; ok {
+		return daemon
+	}
+	return nil
+}
+
+func (engine *DockerTaskEngine) SetDaemonTask(daemonName string, task *apitask.Task) {
+	engine.daemonTasksLock.Lock()
+	defer engine.daemonTasksLock.Unlock()
+	engine.daemonTasks[daemonName] = task
+}
+
+func (engine *DockerTaskEngine) GetDaemonManagers() map[string]dm.DaemonManager {
+	return engine.daemonManagers
```

Comment:
It is not thread-safe for the user of this method to do any read/writes on the returned map. We should return a copy and document that this method returns a copy.

Reply:
But it's not trivial to copy `DaemonManager`... hmmm

Reply:
Since we have a thread-safe `GetDaemonTask` method maybe we can get rid of this method.

Reply:
this is a different map -- we have a 1:1 mapping of daemon:daemonManager.  The daemonManager has a CreateDaemonTask() method which will create the DaemonTask to run.  

Presently this import only happens once as we start up and register capabilities, so thread safety isn't as much of a concern -- I could wrap this in a read lock as it will not be updated after the initial daemon manager import.  In future this could change if we allow for in-place daemon add.

Reply:
ah right this is the managers map. Cool.